### ISSUE TITLE
Tests with priority param are ignored

### ISSUE THREAD

**Issue by nikmazur** [cbeust/testng#1857]
### TestNG Version
6.14.3. Issue started with 6.13, not reproducible on 6.11 or earlier

I have two classes - First where tests have a numbered priority through params, and Second where the params are not set.  I then create a run config to execute all tests in package and run it.

### Expected behavior
All tests in both classes are expected to execute, since by default, if not specified, the priority = 0. 

### Actual behavior
Tests with numbered priority from the FirstClass are ignored while tests without priority from SecondClass are executed.
If I remove 1 of the 3 tests from the FirstClass, all tests from both classes execute.
If I delete or disable tests in SecondClass, tests from FirstClass execute.

### Is the issue reproductible on runner?

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [x] IntelliJ
- [ ] NetBeans

### Test case sample

```java
public class FirstClass {

    @Test (priority = 1)
    private void firsttest() {
        String t = "1";
        assert t.equals("1");
    }

    @Test (priority = 2)
    private void secondtest() {
        String t = "2";
        assert t.equals("2");
    }

    @Test (priority = 3)
    private void thirdtest() {
        String t = "3";
        assert t.equals("3");
    }
}
```
```java
public class SecondClass {

    @Test
    private void fourthtest() {
        String t = "1";
        assert t.equals("1");
    }

    @Test
    private void fifthtest() {
        String t = "2";
        assert t.equals("2");
    }
}
```

**Comment by AFranc1sc0** [comment]
Can confirm I have seen this issue with exact scenario described above. 

**Summary:**
When invoking test classes where some have priority and some do not the tests will priority will not execute.

**Version** 
TestNG 6.14.3
Maven

**Repro steps:** 

1. Create test classes as follows: 
 `public class FirstClass {

    @Test (priority = 1)
    private void firsttest() {
        String t = "1";
        assert t.equals("1");
    }

    @Test (priority = 2)
    private void secondtest() {
        String t = "2";
        assert t.equals("2");
    }

    @Test (priority = 3)
    private void thirdtest() {
        String t = "3";
        assert t.equals("3");
    }
}`

`public class SecondClass {

    @Test
    private void fourthtest() {
        String t = "1";
        assert t.equals("1");
    }

    @Test
    private void fifthtest() {
        String t = "2";
        assert t.equals("2");
    }
}`

2. Create testng xml that calls these methods:
`<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
<suite name="Bug Tester">
	<test name="Bug Test" thread-count="1">
        <classes >
            <class name="FirstClass" />
            <class name="SecondClass" />
        </classes>
	</test>
</suite>`

 3. Execute testng xml 

**Expected:** All requested tests execute.
**Actual:** 
Resulting Command Line Output:
![testexec](https://user-images.githubusercontent.com/32621682/50490067-22ecac80-09d9-11e9-89c0-fc32ce29f44e.png)


**Work around:**

Deleting priority from the tests with priority: 
 `public class FirstClass {

    @Test 
    private void firsttest() {
        String t = "1";
        assert t.equals("1");
    }

    @Test
    private void secondtest() {
        String t = "2";
        assert t.equals("2");
    }

    @Test
    private void thirdtest() {
        String t = "3";
        assert t.equals("3");
    }
}`

`public class SecondClass {

    @Test
    private void fourthtest() {
        String t = "1";
        assert t.equals("1");
    }

    @Test
    private void fifthtest() {
        String t = "2";
        assert t.equals("2");
    }
}`

2. Call the same testng xml

**Results:** 
![workaround](https://user-images.githubusercontent.com/32621682/50490909-9395c800-09dd-11e9-9cef-2bc1e17a850b.png)

**Comment by krmahadevan** [comment]
@AFranc1sc0 - The latest released version of TestNG is `7.0.0-beta1`. Please retry using that.

**Comment by AFranc1sc0** [comment]
@krmahadevan it's fixed! 

Repeated all above tests using latest build 7.0.0-beta1 and it works! 
 

Result: 
![fixed](https://user-images.githubusercontent.com/32621682/51059369-faa8a480-15b9-11e9-9da7-f33c5a0f1298.png)
